packages/test/end-to-end-tests/src/test/oldVersion.ts

```diff
@@ -189,7 +189,7 @@ export function createTestObjectProvider(
             const oldMakeTestLoader = oldProvider.makeTestLoader.bind(oldProvider);
             oldProvider.makeTestLoader = (testContainerConfig?: unknown) => {
                 const testLoader = oldMakeTestLoader(testContainerConfig);
-                loaderContainerTracker.add(testLoader);
+                loaderContainerTracker.add(testLoader as any);
```

Comment:
is there any way we can avoid all these casts (or at least not cast to any)?

Reply:
this is all due to how we currently implement our compat tests, and is a result of dependency confusing between the new and old packages. I have ideas how to fix it. jsut need time to do it.